src/loaders/FileLoader.js

```diff
@@ -122,30 +122,35 @@ class FileLoader extends Loader {
 
 							function readData() {
 
-								reader.read().then( ( { done, value } ) => {
+								reader.read()
+									.then( ( { done, value } ) => {
 
-									if ( done ) {
+										if ( done ) {
 
-										controller.close();
+											controller.close();
 
-									} else {
+										} else {
 
-										loaded += value.byteLength;
+											loaded += value.byteLength;
 
-										const event = new ProgressEvent( 'progress', { lengthComputable, loaded, total } );
-										for ( let i = 0, il = callbacks.length; i < il; i ++ ) {
+											const event = new ProgressEvent( 'progress', { lengthComputable, loaded, total } );
+											for ( let i = 0, il = callbacks.length; i < il; i ++ ) {
 
-											const callback = callbacks[ i ];
-											if ( callback.onProgress ) callback.onProgress( event );
+												const callback = callbacks[ i ];
+												if ( callback.onProgress ) callback.onProgress( event );
+
+											}
+
+											controller.enqueue( value );
+											readData();
 
 										}
 
-										controller.enqueue( value );
-										readData();
+									} ).catch( err => {
 
-									}
+										throw err;
 
```

Comment:
What are you expecting this to do? Throwing the same error you caught is effectively a no op.

Reply:
Apologies, when I was running this locally I thought this was sufficient but it was only because something went wrong with my hot reload.

I've replaced the commit I've had before so hopefully it'll be clearer in what the intent is. I'm open to altering the approach though. I'm not sure if a shared error handler is preferred here, another solution is to switch this to be an async/await so it's easier to have an outer error handle for the recursive promises.